public/src/modules/helpers.js

```diff
@@ -151,6 +152,14 @@
 		return style.join(' ');
 	}
 
+	function showDisplayName(userData) {
+		if (config.showFullnameAsDisplayName && userData.fullname) {
+			return userData.fullname;
+		}
+
+		return userData.username;
+	}
+
```

Comment:
What's the difference between using this helper and using `displayname` in user hash? Is `displayname` not always present?

Reply:
`displayname` is only used on profile pages because we may have both `fullname` and `username` available and we may want to show both of those separately. Rest of the forum uses this helper to determine if we show username or fullname